flink-runtime/src/main/java/org/apache/flink/runtime/io/network/partition/ResultSubpartition.java

```diff
@@ -99,6 +82,23 @@ protected Throwable getFailureCause() {
 
 	abstract public boolean isReleased();
 
+	/**
+	 * Gets the number of non-event buffers in this subpartition.
+	 */
+	abstract public int getBuffersInBacklog();
+
+	/**
+	 * Decreases the number of non-event buffers by one after fetching a non-event
+	 * buffer from this subpartition.
+	 */
+	abstract public void decreaseBuffersInBacklog(Buffer buffer);
+
+	/**
+	 * Increases the number of non-event buffers by one after adding a non-event
+	 * buffer into this subpartition.
+	 */
+	abstract public void increaseBuffersInBacklog(Buffer buffer);
```

Comment:
I'm not quite sure the latter two methods should be in `ResultSubpartition` now since they are quite internal. `increaseBuffersInBacklog()` is only called by `PipelinedSubpartition` and `SpillableSubpartition`. `decreaseBuffersInBacklog()` is (additionally) only by spilled/spillable subpartition views and therefore could be package-private in `SpillableSubpartition` only.

Reply:
The current `parent` in `SpilledSubpartitionView` is `ResultSubpartition` not `SpillableSubpartition`, after replacing the `ResultSubpartition` by `SpillableSubpartition`, we can make these methods package-private as you suggest. I will do that.

Reply:
`package-private`, e.g. `abstract void increaseBuffersInBacklog(Buffer buffer);`, already works without changing anything since `SpilledSubpartitionView` is in the same package

Reply:
Sorry my expression is not correct above. I mean we do not need `decreaseBuffersInBacklog` method in `ResultSubPartition` after modifying the `parent` as `SpillableSubpartition` in  `SpilledSubpartitionView`.

Reply:
yes, that would be nice